src/url.js

```diff
@@ -111,10 +115,15 @@ export function parseUrlDeprecated(url, opt_nocache) {
  * @param {!HTMLAnchorElement} a
  * @param {string} url
  * @param {LruCache=} opt_cache
+ *   Cache is altogether ignored on ESM builds, since an <a> is not used.
  * @return {!Location}
  * @restricted
  */
 export function parseUrlWithA(a, url, opt_cache) {
+  if (IS_ESM) {
+    return /** @type {!Location} */ (new URL(url, a.href));
```

Comment:
There are several callers, which each would have to reset the `a.href` beforehand. I think just doing it here is safest.

Reply:
I see. I was worried these would make it slower, but it looks like the performance properties remain similar ([low hit rate](https://jsbench.me/90kip5jfbm), [high hit rate](https://jsbench.me/fokip5mqfn)).

Reply:
range comparing only old and new, with `a.href = ''`:

|             | cached |    | [low hits](https://jsbench.me/zskip60sh8) |    | [high hits](https://jsbench.me/3gkip6ilpt) |
|-------------|--------|----|------------------------------------------|----|--------------------------------------------|
|             |        |    | _1000 total, 300 unique (100 cached)_    |    | _1000 total, 100 unique (100 cached)_      |
| `<a>`       | yes    |    | 27.4 ops/s ± 40.29%<br>_58.27 % slower_  | 🐇 | 74.58 ops/s ± 1.84%<br>**Fastest**         |
| `URL()`     | no     | 🐇 | 65.65 ops/s ± 0.74%<br>**Fastest**       |    | 67.33 ops/s ± 0.97%<br>_9.72 % slower_     |